lib/cxml/parser.rb

```diff
@@ -45,6 +45,8 @@ def dtd_url
     private
 
     def node_to_hash(node) # rubocop:disable Metrics/AbcSize
```

Comment:
Since Rubocop now complains about this method (https://github.com/officeluv/cxml-ruby/pull/16/checks?check_run_id=1014466537#step:5:46), can we refactor this into something like:

```ruby
    def node_to_hash(node)
      return node.value if node.is_a?(Ox::CData)
      return node if node.is_a? String
      return node.nodes.first if node.nodes.all?(String) && node.attributes.empty?

      transform_node_to_hash node
    end

    def transform_node_to_hash(node)
      hash = node.attributes
      hash.transform_keys!(&method(:underscore_key))
      node.nodes.reduce(hash) do |acc, child_node|
         next acc if child_node.is_a?(Ox::Comment)

         node_hash = {}
         name = child_node.is_a?(String) ? :content : child_node.value
         name = child_node.is_a?(String) || child_node.is_a?(Ox::CData) ? :content : child_node.value
         node_hash[underscore_key(name)] = node_to_hash(child_node)
         acc.merge(node_hash) do |_key, val1, val2|
           [val1, val2].flatten
        end
      end
    end
```